src/main/webapp/assets/js/signUpScript.js

```diff
@@ -52,23 +69,195 @@ async function onLoad() {
   collegeDataList.appendChild(fragment);
   collegeDataList.setAttribute('id', 'colleges');
 
-  // When users select an option from the dropdown, set the college id.
-  document.getElementById('colleges-input-form').addEventListener('change', setCollegeID);
+  // When users submit the form, initiate form validation.
+  emailForm = document.getElementById('email-form');
+  submitFormButton = document.getElementById('form-submit');
+  subscribeFormSubmitted = document.getElementById('subscribe-form-submitted');
+  unsubscribeFormSubmitted = document.getElementById('unsubscribe-form-submitted');
+  submitFormButton.addEventListener('click', submitForm);
+}
+
+/**
+ * On click of the submit button, validate form data and send data to the servlet.
+ * @return {void}
+ */
+async function submitForm() {
+  // Disable multiple submissions.
+  submitFormButton.disabled = true;
+
+  if (validateForm()) {
+    emailForm.action = '/user';
+
+    // Based on user subscribing or unsubscribing, show confirmation that form
+    // was submitted.
+    if (document.getElementById('subscribe').checked) {
+      subscribeFormSubmitted.style.display = 'block';
+    } else {
+      unsubscribeFormSubmitted.style.display = 'block';
+    }
+
+    // Waits for 4000 ms so user can read confirmation and submits the form.
+    setTimeout(function() {
+      emailForm.submit();
+    }, 4000);
```

Comment:
4 seconds is pretty long for a popup - what happens if the user clicks outside the modal and tries to navigate somewhere else before the 4 seconds are up, which causes the form to not actually be submitted?

Reply:
I can switch it to submit the form before the popup shows up and change it to 2 seconds!